diode-server/auth/server.go

```diff
@@ -209,6 +216,13 @@ func (s *Server) introspect(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	err = s.tokenOwnership.HeaderCheck(r.Header, claims)
+	if err != nil {
+		s.logger.Error("failed to validate token ownership with HTTP headers", "error", err)
+		w.WriteHeader(http.StatusForbidden)
+		return
+	}
+
```

Comment:
Tomorrow morning, I'm going to see if there's a good place to add test coverage for these lines in the `introspect` method.